osquery/remote/http_client.cpp

```diff
@@ -382,17 +487,14 @@ Response Client::sendHTTPRequest(Request& req) {
       }
     } catch (std::exception const& /* e */) {
       closeSocket();
-      if (!retry_connect) {
+      if (!retry_connect && ec_ != boost::asio::error::timed_out) {
         retry_connect = true;
       } else {
+        ec_.clear();
         throw;
       }
     }
   } while (true);
-
-  if (!client_options_.keep_alive_) {
```

Comment:
@uptycs-nishant, sorry I this question has been asked before, but why was this keep-alive/close-socket moved into `sendRequest`?

Reply:
There are two ways to get out of function sendHTTPRequest -
1. Via returning the response.
2. Via throwing the exception.

Look at the placement of this check, it is dead code(not reachable).